src/mono/System.Private.CoreLib/src/System/Array.Mono.cs

```diff
@@ -376,37 +377,38 @@ public static unsafe Array CreateInstance(Type elementType, int[] lengths, int[]
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern unsafe void InternalCreate([NotNull] ref Array? result, IntPtr elementType, int rank, int* lengths, int* lowerBounds);
 
-        public object GetValue(int index)
+        private unsafe nint GetFlattenedIndex(ReadOnlySpan<int> indices)
         {
-            if (Rank != 1)
-                ThrowHelper.ThrowArgumentException(ExceptionResource.Arg_Need1DArray);
+            // Checked by the caller
+            Debug.Assert(indices.Length == Rank);
 
-            int lb = GetLowerBound(0);
-            if (index < lb || index > GetUpperBound(0))
-                throw new IndexOutOfRangeException(SR.Argument_IndexOutOfArrayBounds);
-
-            if (GetType().GetElementType()!.IsPointer)
-                throw new NotSupportedException(SR.NotSupported_Type);
-
-            return GetValueImpl(index - lb);
+            nint flattenedIndex = 0;
```

Comment:
The frequent type casting in a bunch of this PR makes me a little anxious, and it's not totally clear to me why it's necessary? Sorry if it's a silly question, but can you explain why the nint stuff is needed?

Reply:
CoreCLR allows allocating multi-dimensional arrays with more than Int32.MaxValue elements. It is why the `nint` is necessary for the CoreCLR and shared part.

From what I can tell, Mono limits the max length to Int32.MaxValue. Is that correct? The use of `nint` here is really just a copy&paste from the CoreCLR version and it should not hurt anything. I can change it to just `int` here if you prefer.

Reply:
I believe Mono does limit the size the way you described. In that case the `nint` should be fine—it just wasn't clear me why it was needed since I didn't realize CoreCLR's limit was different.